cli/command/service/helpers_test.go

```diff
@@ -0,0 +1,45 @@
+package service
+
+import (
+	"bytes"
+	"io/ioutil"
+	"testing"
+
+	"github.com/docker/cli/cli/internal/test"
+	"github.com/spf13/pflag"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWarnDetachDefault(t *testing.T) {
+	var detach bool
+	client := &fakeClient{}
+	cli := test.NewFakeCli(client, ioutil.Discard)
+	flags := pflag.NewFlagSet("test", pflag.ContinueOnError)
+	addDetachFlag(flags, &detach)
+
+	var tests = []struct {
+		detach  bool
+		version string
+
+		expectWarning bool
+	}{
+		{true, "1.28", false},
+		{false, "1.28", false},
+		{false, "1.29", true},
+	}
+
+	for _, test := range tests {
+		out := new(bytes.Buffer)
+		cli.SetErr(out)
+		detach = test.detach
```

Comment:
I don't think this is triggering `flags.Changed()`. You can use `flags.Lookup(flagDetach).Changed = ...`

There's one more test case here, isn't there? `{true, "1.29", false)`, which I think will fail until ^ is fixed